subsys/net/openthread/rpc/server/ot_rpc_message.c

```diff
@@ -37,11 +37,12 @@ ot_msg_key ot_reg_msg_alloc(otMessage *msg)
 
 void ot_msg_free(ot_msg_key key)
 {
-	key--;
-	if (key >= OT_MESSAGES_POOL) {
+	if (key == 0 || key > OT_MESSAGES_POOL) {
```

Comment:
should we check this in `ot_msg_get` as well? If key==0 is passed there for some reason we will go out of bounds

Reply:
Right, I guess the idea was that the integer would underflow and end up greater than `OT_MESSAGES_POOL`, but I thought it's clearer this way. I'll fix get as well.